Events and Logs API (#202)
open-telemetry/oteps: text/0202-events-api.md

# Introducing Events API

> 

Reviewer: nit: remove the extra blank line here

We have an option of adding API for both Logs and Events. However, there is a general consensus that we should not have an API in Otel for creating logs since each language already has multiple logging frameworks. Therefore we restrict the API specification below to Events and call it Events API. For logs, it is recommended that end-users continue to use existing Logging APIs and export the logs into OTLP using the  [appender API](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation-appender-api-internal/src/main/java/io/opentelemetry/instrumentation/api/appender/internal) and LogEmitter SDK. The Events API will offer a subset of the features of LogEmitter SDK and so it will be backed by LogEmitter SDK and the LogRecord data model.

> ## Subtle differences between Logs and Events

Author: updated the text, can you take a look?

> In OpenTelemetry's perspective Log Records and Events are different names for the same concept. However, there are subtle differences in how they are represented in the underlying LogRecord data model. Logs have a mandatory severity level as a first-class parameter that events do not have, and events have a mandatory name that logs do not have. Further, logs typically have messages in string form and events have data in the form of key-value pairs. It is due to this that their API interface requirements are slightly different.

Reviewer: nit: add a blank line between Ln8 and Ln9.

> In OpenTelemetry's perspective Log Records and Events are different names for the same concept. However, there are subtle differences in how they are represented in the underlying LogRecord data model. Logs have a mandatory severity level as a first-class parameter that events do not have, and events have a mandatory name that logs do not have. Further, logs typically have messages in string form and events have data in the form of key-value pairs. It is due to this that their API interface requirements are slightly different.

Reviewer: "Further, logs typically have messages in string form and events have data in the form of key-value pairs." where is this coming from?

Author: Until structured logging, which is relatively new, log messages are typically strings. Event data is more generic and hence best represented using key-value pairs.

## Who requires Events API

Here are a few situations that require recording of Events, there will be more.
* RUM events (Client-side instrumentation)
* Recording kubernetes events
* Recording eBPF events
* Few other event systems described in [example mappings](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#appendix-a-example-mappings) in the data model.

## Can the current Log API interfaces be used for events?

> 1. The log level is fundamental to the Log APIs in almost all the languages; all the methods in the Log interface are named after the log level, and there is usually no generic method to submit a log entry without log level.
> * In JavaScript for Web, the standard method of logging is to use console.log. Events can be created using [Event/CustomEvent](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) interfaces.However, there is no option to define custom destination for these logs and events. Logs go only to console and event listeners are attached to the DOM element that dispatches it.
> * In Android, android.util.Log has methods  Log.v(), Log.d(), Log.i(), Log.w(), and Log.e() to write logs. These methods correspond to the severity level.
> * Swift on iOS has Logger interface that has methods corresponding to severity level too.
> 2. The Log APIs do not have a standard way to pass event attributes. 
> * It may be possible to use the interpolation string args as the parameter to pass event attributes. However, the logging spec seems to map the human readable message (which is obtained after replacing the args in the interpolated string)  to the Body field of LogRecord. 
> * Log4j has an EventLogger interface that can be used to create structured messages with arbitrary key-value pairs, but log4j is not commonly used in Android apps as it is not officially supported on Android as per this [Stack Overflow thread](https://stackoverflow.com/questions/60398799/disable-log4j-jmx-on-android/60407849#60407849) by one of log4j’s maintainers.
> * In python, logging.LogRecord’s extra field is mapped to Otel LogRecord’s attributes but this field is a hidden field and not part of the public interface.

Reviewer: ```suggestion
1. The log level is fundamental to the Log APIs in almost all the languages; all the methods in the Log interface are named after the log level, and there is usually no generic method to submit a log entry without log level.
* In JavaScript for Web, the standard method of logging is to use console.log. Events can be created using [Event/CustomEvent](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) interfaces.However, there is no option to define custom destination for these logs and events. Logs go only to console and event listeners are attached to the DOM element that dispatches it.
* In Android, android.util.Log has methods  Log.v(), Log.d(), Log.i(), Log.w(), and Log.e() to write logs. These methods correspond to the severity level.
* Swift on iOS has Logger interface that has methods corresponding to severity level too.
2. The Log APIs do not have a standard way to pass event attributes. 
* It may be possible to use the interpolation string args as the parameter to pass event attributes. However, the logging spec seems to map the human readable message (which is obtained after replacing the args in the interpolated string)  to the Body field of LogRecord. 
* Log4j has an EventLogger interface that can be used to create structured messages with arbitrary key-value pairs, but log4j is not commonly used in Android apps as it is not officially supported on Android as per this [Stack Overflow thread](https://stackoverflow.com/questions/60398799/disable-log4j-jmx-on-android/60407849#60407849) by one of log4j’s maintainers.
* In python, logging.LogRecord's extra field is mapped to Otel LogRecord's attributes but this field is a hidden field and not part of the public interface.
```

> 1. The log level is fundamental to the Log APIs in almost all the languages; all the methods in the Log interface are named after the log level, and there is usually no generic method to submit a log entry without log level.
> * In JavaScript for Web, the standard method of logging is to use console.log. Events can be created using [Event/CustomEvent](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) interfaces.However, there is no option to define custom destination for these logs and events. Logs go only to console and event listeners are attached to the DOM element that dispatches it.
> * In Android, android.util.Log has methods  Log.v(), Log.d(), Log.i(), Log.w(), and Log.e() to write logs. These methods correspond to the severity level.
> * Swift on iOS has Logger interface that has methods corresponding to severity level too.
> 2. The Log APIs do not have a standard way to pass event attributes. 
> * It may be possible to use the interpolation string args as the parameter to pass event attributes. However, the logging spec seems to map the human readable message (which is obtained after replacing the args in the interpolated string)  to the Body field of LogRecord. 
> * Log4j has an EventLogger interface that can be used to create structured messages with arbitrary key-value pairs, but log4j is not commonly used in Android apps as it is not officially supported on Android as per this [Stack Overflow thread](https://stackoverflow.com/questions/60398799/disable-log4j-jmx-on-android/60407849#60407849) by one of log4j’s maintainers.
> * In python, logging.LogRecord’s extra field is mapped to Otel LogRecord’s attributes but this field is a hidden field and not part of the public interface.

Reviewer: ```suggestion
1. The log level is fundamental to the Log APIs in almost all the languages; all the methods in the Log interface are named after the log level, and there is usually no generic method to submit a log entry without log level.
* In JavaScript for Web, the standard method of logging is to use console.log. Events can be created using [Event/CustomEvent](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) interfaces.However, there is no option to define custom destination for these logs and events. Logs go only to console and event listeners are attached to the DOM element that dispatches it.
* In Android, android.util.Log has methods  Log.v(), Log.d(), Log.i(), Log.w(), and Log.e() to write logs. These methods correspond to the severity level.
* Swift on iOS has Logger interface that has methods corresponding to severity level too.
2. The Log APIs do not have a standard way to pass event attributes. 
* It may be possible to use the interpolation string args as the parameter to pass event attributes. However, the logging spec seems to map the human readable message (which is obtained after replacing the args in the interpolated string)  to the Body field of LogRecord. 
* Log4j has an EventLogger interface that can be used to create structured messages with arbitrary key-value pairs, but log4j is not commonly used in Android apps as it is not officially supported on Android as per this [Stack Overflow thread](https://stackoverflow.com/questions/60398799/disable-log4j-jmx-on-android/60407849#60407849) by one of log4j’s maintainers.
* In python, logging.LogRecord's extra field is mapped to Otel LogRecord's attributes but this field is a hidden field and not part of the public interface.
```

> * In python, logging.LogRecord’s extra field is mapped to Otel LogRecord’s attributes but this field is a hidden field and not part of the public interface.

Reviewer: ```suggestion
* In python, logging.LogRecord's extra field is mapped to Otel LogRecord's attributes but this field is a hidden field and not part of the public interface.
```

> 3. The Log APIs have a message parameter which could map to the Body field of LogRecord. However, this is restricted to String messages and does not allow for a complex structure.

Reviewer: Would you elaborate more here? What is/are "Log APIs" referring to?

Author: I updated the text a bit. The current Log APIs available in the language SDK do not support creating structured logs.

For the above reasons we can conclude that we will need a separate API for generating Events API

## Should OpenTelemetry have an API for logs?

> There’s a general consensus in the Otel community that we should not have a full-fledged logging API unless there is a language that doesn't already have a plethora of logging libraries & APIs to choose from where it might make sense to define one. Further, we will not be able to have the [rich set of configuration options](https://logging.apache.org/log4j/2.x/manual/configuration.html) that some popular logging frameworks provide so the logging API in Otel will only become yet another API.

Reviewer: OTel (uppercase `T`) https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification#acronym

## Events API Interface

For reference, a prototype of the Events API in Java is [here](https://github.com/scheler/opentelemetry-java/pull/1/files)

> Client-side telemetry is one of the initial clients that will use the Events API and so the API will be made available in JavaScript, Java and Swift first to be able to use in the SDKs for Browser, Android and iOS.  It may also be added in Go since there is a kubernetes events receiver implemented in Collector based on Logs data-model.

Reviewer: ```suggestion
Client-side telemetry is one of the initial clients that will use the Events API and so the API will be made available in JavaScript, Java and Swift first to be able to use in the SDKs for Browser, Android and iOS.  It may also be added in Go since there is a Kubernetes events receiver implemented in Collector based on Logs data-model.
```

  * The signature of the method is to be determined by each language and can be overloaded as appropriate. The method MUST record an exception as an Event with the conventions outlined in the exception semantic conventions document. The minimum required argument SHOULD be no more than only an exception object.
  * If RecordException is provided, the method MUST accept an optional parameter to provide any additional event attributes (this SHOULD be done in the same way as for the AddEvent method). If attributes with the same name would be generated by the method already, the additional attributes take precedence.
  * Note: RecordException may be seen as a variant of logEvent with additional exception-specific parameters and all other parameters being optional (because they have defaults from the [exception semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md)).

A LogRecord representing exception event will look like this:
```
{
  time_unix_nano: 1653682410,
  Attributes: {
   event.name: exception,
   exception.type:  OSError,
   exception.message: Division by Zero,

>    exception.stacktrace: “”,

Reviewer: ```suggestion
   exception.stacktrace: "",
```

# Introducing Events and Logs API

> In this OTEP we introduce an Events and Logs API that is based on the OpenTelemetry Log signal. The Events here refer to the independent Events and not to be confused with Span Events which occur only in the context of a span. In OpenTelemetry's perspective Log Records and Events are different names for the same concept - however, there is a subtle difference in how they are represented using the underlying data model that is described below. Although every language has APIs for logs, they are not all capable of creating events. We will describe why the existing Logging APIs are not sufficient for the purpose of creating events.  It will then be evident that we will need an API in OpenTelementry for creating events.

Reviewer 2: >Although every language has APIs for logs, they are not all capable of creating events.

I think this sentence can be misinterpreted and is probably unnecessary. Perhaps delete it.

Author: Deleted.

> We have an option of adding API for both Logs and Events. However, there is a general consensus that we should not have an API in Otel for creating logs since each language already has multiple logging frameworks. Therefore we restrict the API specification below to Events and call it Events API. For logs, it is recommended that end-users continue to use existing Logging APIs and export the logs into OTLP using the  [appender API](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation-appender-api-internal/src/main/java/io/opentelemetry/instrumentation/api/appender/internal) and LogEmitter SDK. The Events API will offer a subset of the features of LogEmitter SDK and so it will be backed by LogEmitter SDK and the LogRecord data model.

Reviewer 2: >Therefore we restrict the API specification below to Events and call it Events API.

This needs some rewording, since we decided to call it "Events and Logs API".

Author: I removed this paragraph entirely and added that the API for Logs is supposed to be only for log appenders.

## Subtle differences between Logs and Events

> In OpenTelemetry's perspective Logs and Events are different names for the same concept. However, there are subtle differences in how they are represented using the underlying LogRecord data model. Logs have a mandatory severity level as a first-class parameter that events do not have, and events have a mandatory name that logs do not have. Further, logs typically have messages in string form and events have data in the form of key-value pairs. It is due to this that their API interface requirements are slightly different.

Reviewer 3: Events here refer to Span Events? If yes, consider linking to the span event spec.

Author: @pmm-sumo removed the redundant text.
@cijothomas The events here refer to independent events, not span events created using the Trace API. I updated the initial paragraph to clarify this.

## Who requires Events API

Here are a few situations that require recording of Events, there will be more.

- RUM events (Client-side instrumentation)
- Recording kubernetes events
- Recording eBPF events

> - Few other event systems described in [example mappings](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#appendix-a-example-mappings) in the data model.

Reviewer 4: I think Profiler events would also fit and should be optionally listed

Author: @MSNev since the profiler events is very new, I will omit it for now. The list here is not meant to be comprehensive.

  - This is a specialized variant of logEvent function, so for anything not specified here, the same requirements as for logEvent apply.
  - The signature of the method is to be determined by each language and can be overloaded as appropriate. The method MUST record an exception as an Event with the conventions outlined in the exception semantic conventions document. The minimum required argument SHOULD be no more than only an exception object.
  - If RecordException is provided, the method MUST accept an optional parameter to provide any additional event attributes (this SHOULD be done in the same way as for the AddEvent method). If attributes with the same name would be generated by the method already, the additional attributes take precedence.
  - Note: RecordException may be seen as a variant of logEvent with additional exception-specific parameters and all other parameters being optional (because they have defaults from the [exception semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md)).

> A LogRecord representing exception event will look like this:

Reviewer 4: nit: `A LogRecord representing exception event may look like this:`
As we haven't defined the conventions for exceptions yet, it would be better to have the language define this as a possible view as opposed to _the_ (will) look like.

Author: @MSNev the conventions for exceptions are already defined, please see - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md

Author: I changed the text to say `may`.

# Introducing Events API

> In this OTEP we introduce an Events API that is based on the OpenTelemetry Log signal. In OpenTelemetry's perspective Log Records and Events are different names for the same concept - however, there is a subtle difference in how they are represented using the underlying data model that is described below. Although every language has APIs for logs, they are not all capable of creating events. We will describe why the existing Logging APIs are not sufficient for the purpose of creating events.  It will then be evident that we will need an API in OpenTelementry for creating events. 

Reviewer 5: > In OpenTelemetry's perspective Log Records and Events are different names for the same concept

Where in the existing spec is this stated?  afaik Events are not a top-level entity, but a sub-entity within Spans.

Author: It is mentioned here - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/overview.md#events-and-logs

## Subtle differences between Logs and Events
In OpenTelemetry's perspective Log Records and Events are different names for the same concept. However, there are subtle differences in how they are represented in the underlying LogRecord data model. Logs have a mandatory severity level as a first-class parameter that events do not have, and events have a mandatory name that logs do not have. Further, logs typically have messages in string form and events have data in the form of key-value pairs. It is due to this that their API interface requirements are slightly different.

## Who requires Events API

Here are a few situations that require recording of Events, there will be more.
* RUM events (Client-side instrumentation)
* Recording kubernetes events
* Recording eBPF events
* Few other event systems described in [example mappings](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#appendix-a-example-mappings) in the data model.

> ## Can the current Log API interfaces be used for events?

Reviewer 6: This list reads to me like the issue is not all languages/runtimes having a structured logging API?

If this is the case it might make more sense to instead have the Log spec say that in those cases Otel logging should provide an API, instead of adding an Events API.

Author: > If this is the case it might make more sense to instead have the Log spec say that in those cases Otel logging should provide an API, instead of adding an Events API.

@tsloughter I want to leave the otep at proposing the Logs API only for log appenders initially. If the language maintainers feel the need for an end-user API for structured logs in OTel, they can add that incrementally later. Let me know what you think.

* In python, logging.LogRecord’s extra field is mapped to Otel LogRecord’s attributes but this field is a hidden field and not part of the public interface.
3. The Log APIs have a message parameter which could map to the Body field of LogRecord. However, this is restricted to String messages and does not allow for a complex structure.

For the above reasons we can conclude that we will need a separate API for generating Events API

## Should OpenTelemetry have an API for logs?

There’s a general consensus in the Otel community that we should not have a full-fledged logging API unless there is a language that doesn't already have a plethora of logging libraries & APIs to choose from where it might make sense to define one. Further, we will not be able to have the [rich set of configuration options](https://logging.apache.org/log4j/2.x/manual/configuration.html) that some popular logging frameworks provide so the logging API in Otel will only become yet another API.

## Events API Interface

> For reference, a prototype of the Events API in Java is [here](https://github.com/scheler/opentelemetry-java/pull/1/files)

Reviewer 7: I propose we go in a slightly different direction that what is described in this document, and what is embodied in the prototype link. 

Here's an alternative prototype, illustrated as [PR](https://github.com/jack-berg/opentelemetry-java/pull/1/files) against the current main branch of otel java. This [test class](https://github.com/jack-berg/opentelemetry-java/pull/1/files#diff-b039e70bb83ee0c5792109dc4fa786c05617f04bebde3d8bde24ce6df26f545e) demonstrates simple usage of the API. These ideas come from conversations in the  Log SIG, which I wanted to make concrete to compare / contrast.

**Summary of differences**
- There is a single API for producing events and log records. NOTE: only the event portion of the API is intended to be used by instrumentation authors.
- Instead of an EventEmitter or Logger (and their corresponding providers) there is a LogEmitter
- There is one method to emit an event called `LogEmitter#eventBuilder(String)`, which accepts the name of the event and returns a EventBuilder for adding additional event details. The `EventBuilder#emit()` method emits the event. 
- There is one method to emit a log record called `LogEmitter#logRecordBuilder()`, which returns a LogRecordBuilder for adding details to log record. The `LogRecordBuilder#emit()` method emits the log record. Note that `EventBuilder` extends `LogRecordBuilder` so they share all the same setters. 

**Reasoning**

There's consensus that we don't want a log API to compete with existing frameworks. The log record API looks remarkably similar to a log API, so why include it instead of just an event API? Well, we need an API to bridge the existing log APIs into OpenTelemetry. That API has a lot of conceptual overlap with an event API. We could add a separate event API and log appender API, but why have all the code and conceptual duplication? Seems better to have a single API that serves both purposes, and to indicate **using documentation** that the log appender API is only meant for appenders, and **NOT** for instrumentation authors.

In otel java, we've bridged two popular logging frameworks ([Log4j](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/OpenTelemetryAppender.java) and  [Logback](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/OpenTelemetryAppender.java)) to otel, but needed to introduce an [appender API](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation-appender-api-internal/src/main/java/io/opentelemetry/instrumentation/api/appender/internal) to decouple the appenders from taking a dependency on the otel java log SDK. The appender API is pretty silly - it just duplicates all the APIs of the otel java log SDK and delegates to a log SDK instance. 

If the API proposed in this otep had a mechanism for producing log records, the otel java appender API would be unnecessary.

Additionally, a single API for producing events as well as log records produces a API / SDK that is very symmetric to the design of tracing / logs. There is no precedent in tracing in logs for a setup where two APIs (events and log appender) delegate to one (or two SDKs) which share the same data model. This symmetry reduces maintenance and cognitive overhead.

Reviewer 7: Here's another prototype, again illustrated as a [PR](https://github.com/jack-berg/opentelemetry-java/pull/2/files) and demonstrated in a [test class](https://github.com/jack-berg/opentelemetry-java/pull/2/files#diff-b039e70bb83ee0c5792109dc4fa786c05617f04bebde3d8bde24ce6df26f545e).

This version has separate APIs for log appenders and events. The naming of classes and required fields reflect the two different use cases.
- On the event side, you have `EventEmitterProvider` and `EventEmitter`. To obtain a `EventEmitter`, you must specify a domain (event names are unique within a domain): `logEmitterProvider.get("my-event-scope", "my-event-domain")`. When emitting an event, you must include an event name: `eventEmitter.eventBuilder("my-event").setAttributes(...).emit()`. 
- On the log side, you have `LoggerProvider` and `Logger`. There is no requirement for an event domain, and there are no required fields when building a log record: `logger.logRecordBuilder().body().setAttributes(...).emit()`. Some subtle differences make it a bit harder to obtain a `Logger` than it is to obtain a `Tracer`, `Meter`, or `EventEmitter`, which is intentional to reflect the fact that the API is only for use by appenders, not instrumentation authors. 

These APIs can be both be implemented by a single Log SDK. This approach has a larger API surface area than my first proposal, but results in an API with more precise language which is arguably less likely to confuse users.

Reviewer 2: I review both prototypes and I think the first approach looks nice (commented on the PR).

Reviewer 8: In the second prototype, the class emitting events is called Logger instead of LogEmitter like in the other prototype. Is this on purpose? Does "Logger" make more sense semantically for the appenders use case?

I am in favor of the second option - it makes it clear that the API is used for events, and it also does not expose the low-level log API to instrumentations libraries.

Reviewer 7: We discussed these options in the java SIG and there was a preference for the second prototype. 

cc @open-telemetry/java-maintainers @open-telemetry/java-instrumentation-maintainers

Author: I am okay with either options but since we have gone back and forth on this a couple times I will wait some time for further comments - on whether we should have one "Logs API" to create both logs and events or have Logs API and Events API separately.

In either case, I assume the API the create application logs will be restricted for use in appenders.

Reviewer 2: We need to circulate this more widely. Tracing/metrics people may have stronger opinions about consistency argument.

Reviewer 8: Here is a second prototype in JS, which has a separate Events API. There is also a draft of Events SDK implementation that is using the Logs SDK
https://github.com/martinkuba/opentelemetry-js/pull/1/files

Author: @MSNev The Span Events created using the Trace API is outside the scope of this OTEP, although I included some text related to it towards the end. The Events API in this OTEP is intended primarily to create Events *outside* the scope of a Span, and so Trace API cannot be used for that purpose.

If you see the section `Comparing with Span Events`, I provided some guidance on how we can extend Trace API to create Span Events using LogRecords. However, that is not intended to be implemented via this OTEP and needs to be taken up separately.

Reviewer 2: > We discussed these options in the java SIG and there was a preference for the second prototype.

@jack-berg can you post what arguments the Java SIG has in favour of doing this?

So far personally for me the consistency argument is very strong and I have not yet seen opposing arguments that are sufficient to warrant having separate APIs.

I do understand that for someone who primarily is interested in Events API, a separate Events API sharply focused on the specific use case is very attractive. However, I think that it is not a good enough argument to warrant losing consistency across OpenTelemetry APIs.

Reviewer 2: @jack-berg @scheler @martinkuba @MSNev @zenmoto @djaglowski 

I started a document to compare the 2 approaches (separate vs one API) side-by-side: https://docs.google.com/document/d/1Uy5L3UThFyc4PelgH51G3OCLFbvzq2xWiNDy2G8iXxU/edit#

Please help me fill the details here.

As agreed in the call we are going to try to make the choice by the next Log SIG. If we don't have a definitive choice then I as a TC member will be the tiebreaker (unless folks don't want tie-breaking and want to make the choice in some other way - I am fine with that).

Reviewer 7: FYI, I've updated the naming of interfaces / classes in the prototype [PR](https://github.com/jack-berg/opentelemetry-java/pull/1/files) that demos a unified API to reflect the naming we discussed in the 6/8 log sig and mentioned by @tigrannajaryan in this [comment](https://github.com/open-telemetry/oteps/pull/202#issuecomment-1150291147).
